### PR TITLE
=doc Consolidate file and resources directives docs #1624

### PR DESCRIPTION
**Pull request by jlprat**
Refs #1624
Creates symlink for directory
Fixes Java inline signatures
Cosmetic fixes

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromBrowseableDirectory.md [hunk 1]
```diff
@@ -19,11 +19,14 @@ Use `getFromBrowseableDirectory` to serve only one directory.
 
 Use `getFromDirectory` if directory browsing isn't required.
 
-For more details refer to [getFromBrowseableDirectory]().
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromBrowseableDirectory.md]: This was a link to the same page, quite useless

File: docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromDirectory.md [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 ## Description
 
-Allows exposing a directory's files for GET requests for its contents.
+Allows exposing a directory's files for `GET` requests for its contents.
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromDirectory.md]: Just a cosmetic fix (it is used with backticks in the other pages)

File: docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromDirectory.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ To serve files from a classpath directory use @ref[getFromResourceDirectory](get
 Note that it's not required to wrap this directive with `get` as this directive will only respond to `GET` requests.
 
 @@@ note
-The file's contents will be read using an Akka Streams *Source* which *automatically uses
+The file's contents will be read using an Akka Streams *`Source`* which *automatically uses
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/getFromDirectory.md]: Also cosmetic

File: docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/index.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # FileAndResourceDirectives
 
 Like the @ref[RouteDirectives](../route-directives/index.md) the @unidoc[FileAndResourceDirectives] are somewhat special in akka-http's routing DSL.
-Contrary to all other directives they do not produce instances of type `Directive[L <: HList]` but rather "plain"
+Contrary to all other directives they do not produce instances of type @scala[`Directive[L <: HList]`]@java[`Directive<L extends HList>`] but rather "plain"
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/index.md]: This is just a translation to Java Syntax

Comment (jrudolph) [reply docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/index.md]: Hmm not sure if that helps Java users :)

Maybe we can cleanup the whole paragraph a bit and say

```
Like the @ref[RouteDirectives](../route-directives/index.md) the @unidoc[FileAndResourceDirectives] are somewhat
special in akka-http's routing DSL. Most other directives wrap an inner route and are therefore used as inner nodes of
the route tree. These directives, instead, are indeed instance of `Route`, i.e.  **leaves** of the route tree that handle a
request themselves without passing it on to an inner route.
```

WDYT?

Comment (jlprat) [reply docs/src/main/paradox/scala/http/routing-dsl/directives/file-and-resource-directives/index.md]: Agree, and I think it didn't really help a subgroup of Scala users either. I like your phrasing, fix is on the way

**Pull request merged**